File: Mbs.Api/Models/Trading/Instruments/Etv.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;
using Mbs.Trading.Currencies;

namespace Mbs.Api.Models.Trading.Instruments
{
    /// <summary>
    /// An additional information for Exchange-Traded Vehicles.
    /// </summary>
    [DataContract]
    public class Etv : IValidatableObject
    {
        /// <summary>
        /// Gets or sets a currency code.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public CurrencyCode Currency { get; set; } = CurrencyCode.Xxx;

        /// <summary>
        /// Gets or sets a trading mode.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string TradingMode { get; set; }

        /// <summary>
        /// Gets or sets all-in fees.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string AllInFees { get; set; }

        /// <summary>
        /// Gets or sets an expence ratio percentage.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string ExpenseRatio { get; set; }

        /// <summary>
        /// Gets or sets a dividend frequency.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string DividendFrequency { get; set; }

        /// <summary>
        /// Gets or sets an issuer.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string Issuer { get; set; }

        /// <summary>
        /// Gets or sets a number of shares outstanding.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public long SharesOutstanding { get; set; }

        /// <inheritdoc />
        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }
}

File: Mbs.Api.Host.Ng/ClientApp/src/app/shared/mbs/instruments/types/etv.ts
import { CurrencyCode } from '../../currencies/currency-code.enum';

/** An additional information for Exchange-Traded Vehicles. */
export class Etv {
  /** A currency code. */
  currency!: CurrencyCode;

  /** A trading mode. */
  tradingMode?: string;

  /** All-in fees. */
  allInFees?: string;

  /** An expence ratio percentage. */
  expenseRatio?: string;

  /** A dividend frequency. */
  dividendFrequency?: string;

  /** An issuer. */
  issuer?: string;

  /** A number of shares outstanding. */
  sharesOutstanding?: number;

  constructor(data?: Etv) {
    if (data) {
      for (const property in data) {
        if (data.hasOwnProperty(property)) {
          (this as any)[property] = (data as any)[property];
        }
      }
    }
  }

  toJSON(data?: any) {
    data = typeof data === 'object' ? data : {};
    // tslint:disable:no-string-literal
    data['Currency'] = this.currency;
    data['TradingMode'] = this.tradingMode;
    data['AllInFees'] = this.allInFees;
    data['ExpenseRatio'] = this.expenseRatio;
    data['DividendFrequency'] = this.dividendFrequency;
    data['Issuer'] = this.issuer;
    data['SharesOutstanding'] = this.sharesOutstanding;
    // tslint:enable:no-string-literal
    return data;
  }
}
